### ISSUE TITLE
Undocumented function with C++ requires clause

### ISSUE THREAD

**Issue by dermojo** [doxygen/doxygen#8911]
**Describe the bug**
Using the following code:
```cpp
#include <concepts>
#include <string>

/// my class
class Foo
{
public:
   /// Ctor for string argument
   template <typename StrType>
   requires std::constructible_from<std::string, StrType> 
   Foo(StrType pathAsStr) : member(pathAsStr) { }

   /// myfunc
   template <typename StrType>
   requires std::constructible_from<std::string, StrType> void myfunc(StrType pathAsStr);
};
```
I get the following warning:
```
def/foo.hpp:15: warning: Member myfunc(StrType pathAsStr) (function) of class Foo is not documented.
```

**Expected behavior**
`myfunc` is documented and the comment should be recognized.

**Version**
1.9.2 

**Additional context**
The problem disappears in various circumstances, e.g. when adding another declaration with comment between constructor and `operator=`, or when removing the constructor.

**Comment by albert-github** [comment]
I think it might be that the `requires` part is not recognized for a constructor.
Quick and nasty workaround (not 100% correct documentation though):
```
   /// Ctor for string argument
   template <typename StrType>
   /// \cond
   requires std::constructible_from<std::string, StrType>
   /// \endcond
   Foo(StrType pathAsStr) : member(pathAsStr) { }
```

**Comment by dermojo** [comment]
Thanks, the workaround fixes this for now.

**Comment by doxygen** [comment]
Please verify if the referenced commit fixes the problem.

**Comment by dermojo** [comment]
I can confirm that the current master doesn't have this problem anymore.

**Comment by dermojo** [comment]
I also patched v1.9.2 with this commit and the problem is gone. Thanks!

**Comment by albert-github** [comment]
Please don't close the issue as this will be done at the moment of an official release

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
